src/loaders/agenda.ts

```diff
@@ -1,42 +1,15 @@
 import * as Agenda from 'agenda';
 import config from '../config';
-import { Collection } from 'mongoose';
 
 export default ({ mongoConnection }) => {
-  const agenda = new Agenda() as any;
+  return new Agenda({
+    mongo: mongoConnection,
+    collection: config.agenda.dbCollection,
+    processEvery: config.agenda.pooltime,
+    maxConcurrency: config.agenda.concurrency,
+  }) as any;
   /**
    * This voodoo magic is proper from agenda.js so I'm not gonna explain too much here.
    * https://github.com/agenda/agenda#mongomongoclientinstance
    */
-  (async () => {
-    await agenda._ready;
-
-    try {
-      (agenda._collection as Collection).ensureIndex(
-        {
-          disabled: 1,
-          lockedAt: 1,
-          name: 1,
-          nextRunAt: 1,
-          priority: -1,
-        },
-        {
-          name: 'findAndLockNextJobIndex',
-        },
-      );
-    } catch (err) {
-      console.log('🔥 Failed to create Agenda index!');
-      console.log(err);
-      throw err;
-    }
-
-    console.log('✌️ Agenda index ensured');
-  })();
-
-  agenda
```

Comment:
This shouldn't be deleted, is the initialization for the agenda jobs

Reply:
Do you mean just the console log or the entire function with `await agenda._ready;`?
I don't think this part is needed because mongo connection is passed.

Reply:
I mean this part

```

agenda.mongo(mongoConnection, config.agenda.dbCollection)	
    .processEvery(config.agenda.pooltime)	
    .maxConcurrency(config.agenda.concurrency);
```
That's what initialize agenda

Reply:
I might be missing something again :)
I replaced the initiation to look more like example from their docs:
https://github.com/agenda/agenda#example-usage

So current code:
```
return new Agenda({
    mongo: mongoConnection,
    collection: config.agenda.dbCollection,
    processEvery: config.agenda.pooltime,
    maxConcurrency: config.agenda.concurrency,
  }) as any;
```
does the same job as before, it initiates agenda, and jobs can be created.
I tested it and works just like before, I'm able to create/remove agenda tasks.

Reply:
Yes, you are right, it does the same 🙌 

I had missed that part when I reviewed the code 😞